### DISCUSSION TITLE
reading network configuration from file

### DISCUSSION THREAD

**Discussion by Naranthiran** [ipxe/ipxe#1040] [General]
Hi all,
I want to save the network configurations(IP address, netmask, gateway and DNS) in the file. So that I can have one common ipxe.iso 

cat and read the input from the file which is stored in the local drive(/boot).

I do not have DHCP server, so I have to configure the IP address manually for each re-installation of operating system(Linux) 

Regards
Naranthiran Duraisamy

**Comment by NiKiZe** [comment]
iPXE can not write files, but in EFI you can load files from filesystem.
Save it as iPXE scripts and you can call it.
There is also autoexec.ipxe which you can have on the iso. In pcbios mode it will run thru initrd= while in EFI it will be detected from disk.

If you want this on local disk but not have iPXE in the same location it will be more challenging.

**Comment by Naranthiran** [comment]
Hi Nikize,

Thanks for your reply.

I have the below grub menu entry on the system which calls the ipxe.efi . The ipxe.efi is embedded with the script.

menuentry "Reinstall via iPXE" {
  insmod part_gpt
  insmod chain
  search --no-floppy --fs-uuid --set=root 84d5ce35-af85-44fe-bd15-c26b8418d0d6
  chainloader /ipxe.efi
}
Embedded script 

#!ipxe
echo Configuring Static IP address for the system ....
ifopen net0
set net0/ip 172.26.127.55
set net0/netmask 255.255.255.0
set net0/gateway 172.26.112.1
set net0/dns 172.26.112.1
chain http://apacheserver/pub/demo/demo.php
boot

I want the static IP address details to be read from the file and used in the ipxe.efi.

I can have both ipxe.efi and file with IP details in /boot.

Can I read the IP address from the file and pass it to ipxe.efi?

Regards
Naranthiran Duraisamy

**Reply by NiKiZe** [reply]
You can not "read" from the file, but you should be able to execute it as a script.

**Reply by Naranthiran** [reply]
Hi Nikize,

Can you share the sample code to execute the script and configure the IP address dynamically?

Regards
Naranthiran Duraisamy

**Reply by NiKiZe** [reply]
You save ipsetup.ipxe or similar next to iPXE binary
The contents would simply be 

   #!ipxe
    set net0/ip 172.26.127.55
    set net0/netmask 255.255.255.0
    set net0/gateway 172.26.112.1
    set net0/dns 172.26.112.1

Script calling this would have
Ifopen net0
chain file://ipsetup.ipxe || echo no ipsetup found

**Reply by Naranthiran** [reply]
Hi Nikize,

I have created the files ipsetup.ipxe and place in the same path were ipxe.efi is available. And trying to call the ipsetup.ipxe files from the script.

But I am getting no ipsetup found. Any idea for the error?

#!ipxe
chain file://ipsetup.ipxe || echo no ipsetup found
chain http://apacheserver/pub/demo/demo.php
boot


Regards
Naranthiran Duraisamy
